### DISCUSSION TITLE
Is there a way to get larastan to detect if there is a property and a method with the same name

### DISCUSSION THREAD

**Discussion by rabol** [larastan/larastan#1751] [Q&A]
Hi

As the subject say: Is it possible for Larastan to detect if a class has a property and a method with the same name?

e.g. this is legal, but not best practice and I would like to get a warning

```php
class foo
{
    public ?string $bar = 'bar';

    public function bar(): string
    {
        return 'test';
    }
}
```

**Comment by canvural** [comment]
Hi,

This is not really related to Larastan itself. But it's about PHPStan. But even PHPStan does not have anything related to this out of the box. You will need to write a custom rule for it.
